activerecord/lib/active_record/transactions.rb

```diff
@@ -234,6 +234,12 @@ def transaction(**options, &block)
         end
       end
 
+      def with_default_isolation_level(isolation_level, &block)
+        with_connection do |connection|
```

Comment:
Eagerly checking out the connection isn't ideal. I think we should try not to do that.

Ultimately we only need to know what the default isolation is when starting the transaction.